### PR TITLE
feat: support `no_std` for `reth-execution-errors`

### LINKED ISSUE #8459: Add no-std support for in reth-storage-errors
### Describe the feature

Ideally, we want to have a subset of crates with no_std supports (evm specifically)
we need to do some prep for this first by making some crates no_std first

## TODO

* add `#![cfg_attr(not(feature = "std"), no_std)]`
* Add to cargo.toml
```
default = ["std"]
std = []
``` 
* Fix errors by manually implementing std::error::Error instead of deriving and put it behind std feature

Best way to do this is `cargo expand` and modify the generated code

### Additional context

_No response_

### LINKED ISSUE #8706: execution erorrs crate no-std support
no_std support for
https://github.com/paradigmxyz/reth/blob/da1536f9c4d995c0168bb5af47b2740e5088439f/crates/evm/execution-errors/Cargo.toml#L2-L2

### PR DESCRIPTION
**Pull request by JackG-eth**
Description

I have created a seperate implementation that uses `thiserror_no_std` instead of manually implementing said traits as `no-std` will be stabilized in core in the not to distant future.

Motivation
closes #8706

### CODE REVIEW

**Comment by JackG-eth** [comment]
@DaniPopes @mattsse

Re our conversation: https://github.com/paradigmxyz/reth/pull/8711

Regarding the comment "You're likely forgetting gating Other(..) variant to also be under cfg(feature = "std"), because it cannot be," I might be wrong, but I believe some crates, like crates/blockchain-tree-api/src/error.rs, will run into trait bound issues. This means we still need to provide manual implementations of the std::error::Error trait where '#[error(transparent)]' is used.

To ensure compatibility with both std and no_std environments, we should conditionally implement std::error::Error for our error types using cfg(feature = "std").

**Review by mattsse — COMMENTED** [review]
okay, so all thiserror-no-std gives us is no manual display and from impl?

is it worth the dependency? given that we already have it in the other pr @DaniPopes ?

**Comment by DaniPopes** [comment]
> To ensure compatibility with both std and no_std environments, we should conditionally implement std::error::Error for our error types using cfg(feature = "std").

I believe you are missing propagating the std feature to thiserror-no-std, as well as it should be declared as default-features=false in the workspace. After this it should be good to merge


> okay, so all thiserror-no-std gives us is no manual display and from impl?
> 

yes

> 
> is it worth the dependency? given that we already have it in the other pr @DaniPopes ?

yes

**Comment by JackG-eth** [comment]
> I believe you are missing propagating the std feature to thiserror-no-std, as well as it should be declared as default-features=false in the workspace. After this it should be good to merge

Yeah you're right, I'll push this tonight and the same for other branches.

**Review by mattsse — APPROVED** [review]
ah nice

File: crates/evm/execution-errors/src/lib.rs
Comment (mattsse) [inline crates/evm/execution-errors/src/lib.rs:1]: this is missing the no_std disable

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/evm/execution-errors/src/lib.rs
Comment (JackG-eth) [reply crates/evm/execution-errors/src/lib.rs:1]: Added and builds succesfully with `no-default-features`

**Review by mattsse — APPROVED** [review]
great, hopefully we can remove the no-std thiserror soon

**Comment by JackG-eth** [comment]
> great, hopefully we can remove the no-std thiserror soon



> great, hopefully we can remove the no-std thiserror soon

Will be happy to revert it once we can

**Pull request merged**